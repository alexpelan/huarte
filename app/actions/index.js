import StateHelper from "../util/StateHelper";
import CONSTS from "../util/Consts";
import api from "../util/Api";

function requestGame(gameId) { 
  return {
    type: "REQUEST_GAME",
    gameId: gameId
  };
}

function receiveGame(json, gameId) {
  const timeLoaded = new Date();
  return {
    type: "RECEIVE_GAME",
    game: json,
    gameId: gameId,
    timeLoaded
  };
}

function requestSeasons() {
  return {
    type: "REQUEST_SEASONS"
  };
}

function receiveSeasons(json) {
  Object.keys(json.seasons).forEach((seasonId) => {
    let season = json.seasons[seasonId];
    season.gamesLoaded = false;
    season.games = [];
  });
  const timeLoaded = new Date();
  return {
    type: "RECEIVE_SEASONS",
    seasons: json.seasons,
    timeLoaded
  };
}

function requestGameList() {
  return {
    type: "REQUEST_GAME_LIST",
  };
}

function receiveGameList(json, seasonId) {
  Object.keys(json.games).forEach((gameId) => {
    let game = json.games[gameId];
    game.loaded = false;
  });
  const timeLoaded = new Date();
  return {
    type: "RECEIVE_GAME_LIST",
    games: json.games,
    seasonId, 
    timeLoaded
  };
}

function selectQuestion(round, categoryIndex, clueIndex){
  return {
    type: "SELECT_QUESTION",
    round: round,
    categoryIndex: categoryIndex,
    clueIndex: clueIndex
  };
};

function updateScore(delta, wasCorrect) {
  return {
    type: "UPDATE_SCORE",
    delta: delta
  };
};

function nextRound(currentRound) {
  var nextRound;
  if (currentRound === "jeopardy") {
    nextRound = "double_jeopardy";
  } else if (currentRound === "double_jeopardy") {
    nextRound = "final_jeopardy";
  } else if (currentRound === "final_jeopardy") {
    Common.saveSingleGameStatistics(StateHelper.getCurrentGame());
  }

  return {
    type: "NEXT_ROUND",
    nextRound: nextRound
  };
}

function bidFinalJeopardy(bid) {
  return {
     type: "BID_FINAL_JEOPARDY",
     bid: bid
  };
}

function noop() {
  return {
    type: "NOOP"
  };
}

//*************************
// THUNKS
//*************************
function fetchGameList(store, seasonId) {
  if (StateHelper.isSeasonLoaded(store, seasonId) && StateHelper.isCacheValid(store.getState().seasons[seasonId].timeLoaded, CONSTS.MILLISECONDS_IN_DAY)) {
    return noop();
  }

  return function(dispatch) {
    dispatch(requestGameList());

    return api.fetchGameList(seasonId)
      .then((response) => response.json())
      .then((json) => {
        dispatch(receiveGameList(json, seasonId));
      });
  };
}

function fetchSeasons(store) {
  if (StateHelper.isCacheValid(store.getState().seasonsLoadedTime, CONSTS.MILLISECONDS_IN_DAY)) {
    return noop();
  }

  return function(dispatch) {
    dispatch(requestSeasons());

    return fetch(CONSTS.SEASONS_REQUEST_URL)
      .then((response) => response.json())
      .then((json) => {
        dispatch(receiveSeasons(json));
      });
  };
}

function fetchGame(store, gameId) {
  if (store.getState().games[gameId].loaded && StateHelper.isCacheValid(store.getState().games[gameId].timeLoaded, CONSTS.MILLISECONDS_IN_DAY)) {
    return noop();
  }

  return function(dispatch) {
    //first dispatch that we are requesting
    dispatch(requestGame(gameId));

    //return a promise 
    return  fetch(CONSTS.GAME_REQUEST_URL + gameId)
      .then((response) => response.json())
      .then((json) => {
        dispatch(receiveGame(json, gameId));
      });

  };  
}

export {
  requestGame,
  receiveGame,
  requestSeasons,
  receiveSeasons,
  requestGameList,
  receiveGameList,
  selectQuestion,
  updateScore,
  nextRound,
  bidFinalJeopardy,
  noop,
  fetchGame,
  fetchGameList,
  fetchSeasons
};