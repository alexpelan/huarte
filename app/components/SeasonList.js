import React from "react";
import {
  StatusBar,
  ListView,
  Text
} from "react-native";

import GameList from "./GameList";
import SimpleMessage from "./SimpleMessage";

import {
  fetchSeasons
} from "../actions/index";

import StateHelper from "../util/StateHelper";
import styles from "../styles/styles";

const SeasonList = React.createClass({
   getInitialState: function() {
    var dataSource = new ListView.DataSource({
        rowHasChanged: (row1, row2) => row1 !== row2
    });
    return {
      dataSource: dataSource.cloneWithRows([])
    };
  },

  componentDidMount: function() {
    const store = this.props.store;
    this.unsubscribe = store.subscribe(() => {
      var state = store.getState();
      if (state.seasonsLoaded) {
        this.setState({
          dataSource: this.state.dataSource.cloneWithRows(StateHelper.transformObjectToListDataSource(state.seasons))
        });
      }
    });
    store.dispatch(fetchSeasons(store));
  },

  componentWillUnmount: function() {
    this.unsubscribe();
  },

  selectSeason: function(season) {
    const store = this.props.store;
    this.props.navigator.push({
      title: season.displayName,
      component: GameList,
      passProps: {
        season,
        store
      },
    });
  },

  render: function() {
    StatusBar.setBarStyle('default', true);
    if(!this.props.store.getState().seasonsLoaded) {
      return (
          <SimpleMessage></SimpleMessage>
      );
    }
    return (
      <ListView
        dataSource={this.state.dataSource}
        renderRow={(season, sectionID) => this.renderSeason(season)}
        automaticallyAdjustContentInsets={false} // ????? https://github.com/facebook/react-native/issues/721
        style={styles.listView}/>
    )
  },

  renderSeason: function(season) {
    return (
      <Text style={styles.listItem}
        onPress={() => this.selectSeason(season)}>
        {season.displayName}
      </Text>
    );
  }
});

export default SeasonList;