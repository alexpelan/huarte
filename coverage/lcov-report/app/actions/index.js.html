<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for app/actions/index.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../prettify.css" />
    <link rel="stylesheet" href="../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../index.html">All files</a> / <a href="index.html">app/actions</a> index.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">68.75% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>33/48</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">62.5% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>10/16</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">67.86% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>19/28</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">68.75% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>33/48</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62
63
64
65
66
67
68
69
70
71
72
73
74
75
76
77
78
79
80
81
82
83
84
85
86
87
88
89
90
91
92
93
94
95
96
97
98
99
100
101
102
103
104
105
106
107
108
109
110
111
112
113
114
115
116
117
118
119
120
121
122
123
124
125
126
127
128
129
130
131
132
133
134
135
136
137
138
139
140
141
142
143
144
145
146
147
148
149
150
151
152
153
154
155
156
157
158
159
160
161
162
163
164
165
166
167
168
169
170
171
172
173
174
175
176
177
178
179
180
181
182
183
184
185
186
187
188
189
190
191
192
193
194
195
196
197
198
199</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-yes">5x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">45x</span>
<span class="cline-any cline-yes">45x</span>
<span class="cline-any cline-yes">45x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-yes">9x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">847x</span>
<span class="cline-any cline-yes">847x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-yes">7x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">2x</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import StateHelper from "../util/StateHelper";
import CONSTS from "../util/Consts";
import api from "../util/Api";
&nbsp;
function requestGame(gameId) { 
  return {
    type: "REQUEST_GAME",
    gameId: gameId
  };
}
&nbsp;
function receiveGame(json, gameId) {
  const timeLoaded = new Date();
  return {
    type: "RECEIVE_GAME",
    game: json,
    gameId: gameId,
    timeLoaded
  };
}
&nbsp;
function selectGame(game) {
  return {
    type: "SELECT_GAME",
    gameId: game.id
  };
}
&nbsp;
function requestSeasons() {
  return {
    type: "REQUEST_SEASONS"
  };
}
&nbsp;
function receiveSeasons(json) {
  Object.keys(json.seasons).forEach((seasonId) =&gt; {
    let season = json.seasons[seasonId];
    season.gamesLoaded = false;
    season.games = [];
  });
  const timeLoaded = new Date();
  return {
    type: "RECEIVE_SEASONS",
    seasons: json.seasons,
    timeLoaded
  };
}
&nbsp;
function requestGameList() {
  return {
    type: "REQUEST_GAME_LIST",
  };
}
&nbsp;
function receiveGameList(json, seasonId) {
  Object.keys(json.games).forEach((gameId) =&gt; {
    let game = json.games[gameId];
    game.loaded = false;
  });
  const timeLoaded = new Date();
  return {
    type: "RECEIVE_GAME_LIST",
    games: json.games,
    seasonId, 
    timeLoaded
  };
}
&nbsp;
function <span class="fstat-no" title="function not covered" >selectQuestion(</span>round, categoryIndex, clueIndex){
<span class="cstat-no" title="statement not covered" >  return {</span>
    type: "SELECT_QUESTION",
    round: round,
    categoryIndex: categoryIndex,
    clueIndex: clueIndex
  };
};
&nbsp;
function <span class="fstat-no" title="function not covered" >updateScore(</span>delta, wasCorrect) {
<span class="cstat-no" title="statement not covered" >  return {</span>
    type: "UPDATE_SCORE",
    delta: delta
  };
};
&nbsp;
function <span class="fstat-no" title="function not covered" >nextRound(</span>currentRound) {
  var nextRound;
<span class="cstat-no" title="statement not covered" >  if (currentRound === "jeopardy") {</span>
<span class="cstat-no" title="statement not covered" >    nextRound = "double_jeopardy";</span>
  } else <span class="cstat-no" title="statement not covered" >if (currentRound === "double_jeopardy") {</span>
<span class="cstat-no" title="statement not covered" >    nextRound = "final_jeopardy";</span>
  } else <span class="cstat-no" title="statement not covered" >if (currentRound === "final_jeopardy") {</span>
<span class="cstat-no" title="statement not covered" >    Common.saveSingleGameStatistics(StateHelper.getCurrentGame());</span>
  }
&nbsp;
<span class="cstat-no" title="statement not covered" >  return {</span>
    type: "NEXT_ROUND",
    nextRound: nextRound
  };
}
&nbsp;
function <span class="fstat-no" title="function not covered" >bidFinalJeopardy(</span>bid) {
<span class="cstat-no" title="statement not covered" >  return {</span>
     type: "BID_FINAL_JEOPARDY",
     bid: bid
  };
}
&nbsp;
function <span class="fstat-no" title="function not covered" >setError(</span>error) {
<span class="cstat-no" title="statement not covered" >  return {</span>
    type: "GLOBAL_SET_ERROR",
    error: error
  };
}
&nbsp;
function <span class="fstat-no" title="function not covered" >dismissError(</span>) {
<span class="cstat-no" title="statement not covered" >  return {</span>
    type: "GLOBAL_DISMISS_ERROR"
  };
}
&nbsp;
function noop() {
  return {
    type: "NOOP"
  };
}
&nbsp;
//*************************
// THUNKS
//*************************
function fetchGameList(store, seasonId) {
  if (StateHelper.isSeasonLoaded(store, seasonId) &amp;&amp; StateHelper.isCacheValid(StateHelper.getSeasonById(store, seasonId).timeLoaded, CONSTS.MILLISECONDS_IN_DAY)) {
    return noop();
  }
&nbsp;
  return function(dispatch) {
    dispatch(requestGameList());
&nbsp;
    return api.fetchGameList(seasonId, store)
      .then(<span class="fstat-no" title="function not covered" >(j</span>son) =&gt; {
<span class="cstat-no" title="statement not covered" >        dispatch(receiveGameList(json, seasonId));</span>
      })
      .catch(() =&gt; {}); //FIXFIX: better way to handle this than empty catch? We handle any errors in api.js
  };
}
&nbsp;
function fetchSeasons(store) {
  if (StateHelper.isCacheValid(store.getState().seasonsLoadedTime, CONSTS.MILLISECONDS_IN_DAY)) {
    return noop();
  }
&nbsp;
  return function(dispatch) {
    dispatch(requestSeasons());
&nbsp;
    return api.fetchSeasonsList(store)
      .then(<span class="fstat-no" title="function not covered" >(j</span>son) =&gt; {
<span class="cstat-no" title="statement not covered" >        dispatch(receiveSeasons(json));</span>
      })
      .catch(() =&gt; {});
  };
}
&nbsp;
function fetchGame(store, gameId) {
  if (store.getState().games[gameId].loaded &amp;&amp; StateHelper.isCacheValid(store.getState().games[gameId].timeLoaded, CONSTS.MILLISECONDS_IN_DAY)) {
    return noop();
  }
&nbsp;
  return function(dispatch) {
    //first dispatch that we are requesting
    dispatch(requestGame(gameId));
&nbsp;
    //return a promise 
    return  api.fetchGame(gameId, store)
      .then(<span class="fstat-no" title="function not covered" >(j</span>son) =&gt; {
<span class="cstat-no" title="statement not covered" >        dispatch(receiveGame(json, gameId));</span>
      })
      .catch(() =&gt; {});
&nbsp;
  };  
}
&nbsp;
export {
  requestGame,
  receiveGame,
  selectGame,
  requestSeasons,
  receiveSeasons,
  requestGameList,
  receiveGameList,
  selectQuestion,
  updateScore,
  nextRound,
  bidFinalJeopardy,
  setError,
  dismissError,
  noop,
  fetchGame,
  fetchGameList,
  fetchSeasons
};</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Wed Jul 26 2017 23:57:22 GMT-0400 (EDT)
</div>
</div>
<script src="../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../sorter.js"></script>
</body>
</html>
