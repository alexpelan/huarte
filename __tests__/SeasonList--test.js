import "react-native";
import React from "react";
import SeasonList from "../app/components/SeasonList";
import Factory from "../testHelpers/Factory";
import {receiveSeasons} from "../app/actions/index";

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock("../app/util/Api");

test('initially renders a loading screen', () => {
	let store = Factory.createNewStore();

	const tree = renderer.create(
		<SeasonList store={store} />
	).toJSON();
	expect(tree).toMatchSnapshot();
});

test('renders a list of seasons after the seasons have been received', () => {
	let store = Factory.createNewStore();
	const seasonList = Factory.seasonList();
	store.dispatch(receiveSeasons(seasonList));

	const tree = renderer.create(
		<SeasonList store={store} />
	).toJSON();
	expect(tree).toMatchSnapshot();
});