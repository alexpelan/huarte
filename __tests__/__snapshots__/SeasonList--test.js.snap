// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`initially renders a loading screen 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#0000af",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "color": "white",
        "fontSize": 24,
      }
    }
  >
    Loading...
  </Text>
</View>
`;

exports[`renders a list of seasons after the seasons have been received 1`] = `
<RCTScrollView
  automaticallyAdjustContentInsets={false}
  dataSource={
    ListViewDataSource {
      "items": 5,
    }
  }
  renderRow={[Function]}
  renderScrollComponent={[Function]}
  style={
    Object {
      "backgroundColor": "#0000af",
      "paddingTop": 70,
    }
  }
>
  <View>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Season 33
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Season 32
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Season 31
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Season 30
    </Text>
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
      onPress={[Function]}
      style={
        Object {
          "color": "white",
          "fontSize": 20,
          "marginBottom": 8,
          "textAlign": "center",
        }
      }
    >
      Season 29
    </Text>
  </View>
</RCTScrollView>
`;
